### PR TITLE
remove versionless error output after successful resolution

### PR DESCRIPTION
**Pull request by rsherget**
Versionless features used with platform environment variable resolves without errors

### CODE REVIEW

**Comment by rsherget** [comment]
#build

**Comment by rsherget** [comment]
#build

**Comment by rsherget** [comment]
Combined with code from #29111

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java [hunk 1]
```diff
@@ -2041,21 +2041,37 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
...
+                        compatibility = featureRepository.getCompatibilityFeature(platform);
+                        if(compatibility != null){
+                            break;
+                        }
+                    }
+                }
+                if(compatibility == null){
+                    break;
+                }
+                compatibility = featureRepository.getCompatibilityFeature(platforms.toArray()[0].toString());
```
Comment (cbridgha) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java]: This line worries me - I did add some null checks over the weekend - should we add here?

**Review by jimblye — DISMISSED** [review]

**Review by tbitonti — DISMISSED** [review]

**Comment by rsherget** [comment]
#build

**Review by cbridgha — DISMISSED** [review]
Thanks looks good

**Comment by rsherget** [comment]
#build

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java [hunk 2]
```diff
@@ -2038,33 +2038,41 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
...
+            }
+            if (compatibility == null) {
+                continue; // Compatibility features are not installed.  The image is incomplete.
+            }
 
-                if(compatibleWithPlatform){
-                    unresolvedVersionless.add(getFeatureName(versionlessResolved.getKey()));
+            String compatibilityBaseName = featureRepository.getFeatureBaseName(compatibility.getFeatureName());
+            for(String resolvedPlat : result.getResolvedPlatforms()){
+                String resolvedBaseName = featureRepository.getFeatureBaseName(featureRepository.getCompatibilityFeature(resolvedPlat).getFeatureName());
```
Comment (cbridgha) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java]: Can you do a null check here?  Sorry but think the repo could be minified...  and not all the compat features are there?   this is what we ran into on the weekend

**Review by cbridgha — DISMISSED** [review]
looks good

**Comment by rsherget** [comment]
#build

**Comment by rsherget** [comment]
#build

**Review by cbridgha — DISMISSED** [review]
Looks good

**Comment by rsherget** [comment]
#build

**Review by jimblye — APPROVED** [review]

**Comment by cbridgha** [comment]
Build finished with 3 unrelated and known issues...   merging...

**Pull request merged**